crates/runtime/src/embeddings/index/s3/index.rs

```diff
@@ -197,6 +197,40 @@ impl VectorIndex for S3Vector {
     }
 }
 
+impl VectorIndex for S3Vector {
+    /// Use a [`S3VectorsListTable`] and then:
+    ///   1. Convert the primary key to its appropriate name and data type
+    ///   2. Rename [`S3_VECTOR_EMBEDDING_NAME`] appropriately
+    fn list_table_provider(
+        &self,
+    ) -> Result<Arc<dyn TableProvider>, Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
This pattern makes me believe we should really be returning a LogicalPlan here, not a TableProvider. The LogicalPlan could encode the projection without needing to wrap the TableProvider

Reply:
That could be a better design. I had not yet figured out how to reconcile (eventually) `EmbeddingTable` (i.e. embeddings JIT / accelerated) with the pattern we have for index-based vector search. `LogicalPlan` makes more sense here.